### PR TITLE
Don't suggest replacing `builtin.open()` with `Path.open()` if the latter doesn't support all options

### LINKED ISSUE #7620: PTH123 suggest to use pathlib.Path.open() which does not implement opener
### Issue
In case an `opener` is used into `open()` function, for creating a file with certain permission for instance, ruff `PTH123` suggests to use `pathlib.Path.open()` which does not implement the `opener`.

```py
import os
dir_fd = os.open('somedir', os.O_RDONLY)
def opener(path, flags):
    return os.open(path, flags, dir_fd=dir_fd)

with open('spamspam.txt', 'w', opener=opener) as f:
    print('This will be written to somedir/spamspam.txt', file=f)
``` 

- ruff version range: v0.0.276 - v0.0.291

### Reference
- [ruff PTH123](https://docs.astral.sh/ruff/rules/builtin-open/)
- [python `open()`](https://docs.python.org/3/library/functions.html#open)
- [python `pathlib.Path.open()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.open)

### Solution
- Ruff can stop reporting it in case an `opener` is defined
- Using `# noqa: PTH123`
- …

### PR DESCRIPTION
**Pull request by konstin**
**Summary** Check that `closefd` and `opener` aren't being used with `builtin.open()` before suggesting `Path.open()` because pathlib doesn't support these arguments.

Closes #7620

**Test Plan** New cases in the fixture.

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs [hunk 1]
```diff
@@ -86,7 +86,39 @@ pub(crate) fn replaceable_by_pathlib(checker: &mut Checker, expr: &Expr) {
...
+                    //              ^^^^         ^^^^
+                    // ```
+                    if call
+                        .arguments
+                        .find_argument("closefd", 6)
+                        .is_some_and(|expr| {
+                            !expr.as_constant_expr().is_some_and(|constant| {
+                                matches!(constant.value, Constant::Bool(true))
+                            })
+                        })
```
Comment (konstin) [inline crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs:108]: Do we have a `.is_default_value("closefd", 6, Constant::Bool(true))` helper? If no, should i create one?

File: crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs [hunk 2]
```diff
@@ -86,7 +86,39 @@ pub(crate) fn replaceable_by_pathlib(checker: &mut Checker, expr: &Expr) {
...
+                    //      closefd=True, opener=None)
+                    //              ^^^^         ^^^^
+                    // ```
+                    if call
+                        .arguments
+                        .find_argument("closefd", 6)
+                        .is_some_and(|expr| {
+                            !expr.as_constant_expr().is_some_and(|constant| {
+                                matches!(constant.value, Constant::Bool(true))
+                            })
```
Comment (charliermarsh) [inline crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs:107]: Can these be done with a single match? Like:

```
matches!(expr, Expr::Constant(ast::ExprConstant { value: Constant::Bool(true), .. }))
```

File: crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs:108]: I think what you have here is fine personally.

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
Nice, thanks. Does `open` support all the other kwargs?

**Pull request merged**